From 2c01fb98f8587a38d0c5bc8ede008d6f0b8e8005 Mon Sep 17 00:00:00 2001
From: David Runge <dave@sleepmap.de>
Date: Wed, 8 Dec 2021 13:55:33 +0100
Subject: [PATCH] Raise when datetime attribute of a Date: header is None

hyperkitty/management/commands/hyperkitty_import.py:
Change `DbImporter._get_date()` to raise when the retrieved `Date:`
header's `datetime` attribute is `None`.
This is required as python 3.10.0 introduces a change, that may return
invalid string representations of a `Date:` header, while setting its
`datetime` attribute to `None`:

https://github.com/python/cpython/commit/303aac8c56609290e122eecc14c038e9b1e4174a
---
 hyperkitty/management/commands/hyperkitty_import.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hyperkitty/management/commands/hyperkitty_import.py b/hyperkitty/management/commands/hyperkitty_import.py
index 48dd248d..baae013d 100644
--- a/hyperkitty/management/commands/hyperkitty_import.py
+++ b/hyperkitty/management/commands/hyperkitty_import.py
@@ -134,6 +134,7 @@ class DbImporter(object):
             date = message.get(header)
             if date:
                 assert all(char in printable for char in date)
+                assert date.datetime is not None
         except (AssertionError, TypeError, ValueError) as e:
             if self.verbose:
                 self.stderr.write(
-- 
GitLab

